core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java

```diff
@@ -106,8 +107,14 @@
  * 		<li>Make sure your application is able to reach <code>https://login.microsoftonline.com</code>. Required for token retrieval. </li>
  * </ol>
  *
+ * To make use of a multi-mailbox situation:
+ * Specify the target mailbox in front of your folder name, separated by 'mailboxFolderSeparator' which defaults to '|', for example <code>"example@domain.com|My folder"</code>
+ * Please consider the following behavioural differences when making using:
  *
- * N.B. MS Exchange is susceptible to problems with invalid XML characters, like &amp;#x3;. 
+ * - copyFile - When copying an email message from mailbox A to mailbox B an expensive call to service.bind() is done to retrieve the newly created object in mailbox B.
+ * - moveFile - When moving an email message from mailbox A to mailbox B, null will be returned
+ *
```

Comment:
Hoe denken jullie over de volgende voorstellen?

- Copy File
Er wordt nu automatisch een 'bind' gedaan wat een expensive call kan zijn, is het een idee om een attribuut te introduceren waarmee de frank ontwikkelaar kan aangeven of hij/zij geinteresseerd is in het object dat aangemaakt wordt. Zo kan de mogelijk dure call voorkomen worden.

- Move File
Er is nu geen goede manier voor de frank ontwikkelaar om een file te moven tussen mailboxen en daarbij het nieuwe object terug te krijgen aangezien er null vanuit de service komt. In de test class heb ik hier omheen gewerkt door een reference te setten en vervolgens op reference de e-mail terug te vinden. Op dit moment is het niet mogelijk voor een frank ontwikkelaar om een reference te setten en obv hiervan vervolgens met een list dit item terug te vinden. Wellicht kunnen we een attribuut introduceren waarmee er onder water een reference + list wordt gedaan voor de frank ontwikkelaar. Een andere optie is om de frank ontwikkelaar zelf een reference te laten setten middels en zelf een list te laten doen, kan opzich ook geen kwaad voor het geval dat de ontwikkelaar slechts een reference wilt setten.

Reply:
Ik denk dat om dit op te lossen het beste IBasicFileSystem als volgt aangepast kan worden:
```
	/**
	 * Moves the file to a another folder.
	 * Does not need to check for existence of the source or non-existence of the destination.
	 * Returns the moved file, or null if no file was moved or there is no reference to the moved file.<br/>
	 * If the reference to the moved file might be unknown after the move, then a call to
	 * {@link #resultantKnownAfterMove(Object, String)} must return false; <br/>
	 * If the reference to the moved file is unknown after the move, and <code>resultantMustBeReturned</code> is set, 
	 * then an Exception must be thrown; 
	 */
	public F moveFile(F f, String destinationFolder, boolean createFolder, boolean resultantMustBeReturned) throws FileSystemException;
	public boolean resultantKnownAfterMove(F f, String destinationFolder) throws FileSystemException;
	/**
	 * Copies the file to a another folder.
	 * Does not need to check for existence of the source or non-existence of the destination.
	 * Returns the copied file, or null if no file was copied or there is no reference to the copied file.
	 * If the reference to the copied file might be unknown after the copy, then a call to
	 * {@link #resultantKnownAfterCopy(Object, String)} must return false; <br/>
	 * If the reference to the copied file is unknown after the copy, and <code>resultantMustBeReturned</code> is set, 
	 * then an Exception must be thrown; 
	 */
	public F copyFile(F f, String destinationFolder, boolean createFolder, boolean resultantMustBeReturned) throws FileSystemException;
	public boolean resultantKnownAfterCopy(F f, String destinationFolder) throws FileSystemException;

```